src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -3913,20 +3913,20 @@ def onCopyToClipboard(self, format=None):
             }
             file_path = save_dialog.getSaveFileName(**kwargs)
             filename = file_path[0]
-
-            if file_path[1] == 'Text (*.txt)':
-                Type_output = Text_output
-                filename = '.'.join((filename, 'txt'))
-            elif file_path[1] == 'Excel (*.xls)':
-                Type_output = Excel_output
-                filename = '.'.join((filename, 'xls'))
-            elif file_path[1] == 'Latex (*.log)':
-                Type_output = Latex_output
-                filename = '.'.join((filename, 'log'))
-
-            file_open = open(filename, 'w')
-            with file_open:
-                file_open.write(Type_output)
+            if filename:
```

Comment:
My suggestion would be to immediately return if no filename is given `if not filename: return`. The offered implementation suggests potential use cases and code execution if no filename is given.